hopp/type_dec.py

```diff
@@ -35,17 +35,20 @@
 
 ### Custom callables for attrs objects and functions
 
-def hopp_array_converter(data: Iterable) -> np.ndarray:
-    try:
-        a = np.array(data, dtype=hopp_float_type)
-    except TypeError as e:
-        raise TypeError(e.args[0] + f". Data given: {data}")
-    return a
+def hopp_array_converter(dtype: Any = hopp_float_type) -> Callable:
+    def converter(data: Iterable):
```

Comment:
What is the reason to return a converter?

Reply:
Forgot to add one before, but just added a docstring. It's a function that returns a converter for any NumPy `dtype` so we don't have to make new converter functions